go/test/endtoend/vtgate/foreign_keys/fk_test.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package foreign_keys
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"vitess.io/vitess/go/test/endtoend/utils"
+)
+
+// TestInsertions tests that insertions work as expected when foreign key management is enabled in Vitess.
+func TestInsertions(t *testing.T) {
+	mcmp, closer := start(t)
```

Comment:
I suspect we won't be using the `mcmp` functionality very much, and then we might as well limit the amount of startup we do in this test. let's minimise the test setup to only what we need